components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java

```diff
@@ -169,4 +171,71 @@ private static Instant parseDateTime(String dateTimeString) {
         }
         return localDateTime;
     }
+
+    /**
+     * Util function to mask content.
+     *
+     * @param content Content that needs to be masked.
+     * @return masked content.
+     */
+    public static String maskContent(String content) {
+
+        if (StringUtils.isNotEmpty(content)) {
+            Pattern pattern = Pattern.compile(LogConstants.CONTENT_MASKING_REGEX);
+            content = pattern.matcher(content).replaceAll(LogConstants.MASKING_CHARACTER);
+        }
+        return content;
+    }
+
+    /**
+     * Util function to mask claim values except userid claim.
+     *
+     * @param claims Map of user claims.
+     * @return sanitized map of user claims.
+     */
+    public static Map<String, String> getMaskedClaimsMap(Map<String, String> claims) {
+
+        Map<String, String> sanitizedClaims = new HashMap<>();
+        if (MapUtils.isNotEmpty(claims)) {
+            for (Map.Entry<String, String> entry : claims.entrySet()) {
+                if (entry.getKey().equals(LogConstants.userIdClaimURI)) {
+                    sanitizedClaims.put(entry.getKey(), entry.getValue());
+                } else {
+                    sanitizedClaims.put(entry.getKey(), maskContent(entry.getValue()));
+                }
+            }
+        }
+        return sanitizedClaims;
```

Comment:
we can refactor this

claims.computeIfPresent(LogConstants.userIdClaimURI, (k, v) -> maskContent(v));

Reply:
Here we are masking all the claim values except for userId. Therefore computeIfPresent method cannot be applied here.